### PR TITLE
[Descartes_Planning_and_Execution] Fixed Eigen matrix initialization error 

### PR DESCRIPTION
**Pull request by ngwk1**
Changed the matrix for `rot` from 3x3 to 4x4 to overcome **Eigen Initialization Error**.

### CODE REVIEW

File: exercises/Descartes_Planning_and_Execution/solution_ws/src/plan_and_run/src/demo_application.cpp [hunk 1]
```diff
@@ -202,9 +202,10 @@ bool DemoApplication::createLemniscateCurve(double foci_distance, double sphere_
       unit_y = (unit_z .cross(unit_x)).normalized();
 
       Eigen::Isometry3d rot;
-      rot.matrix() << unit_x(0),unit_y(0),unit_z(0)
-         ,unit_x(1),unit_y(1),unit_z(1)
-         ,unit_x(2),unit_y(2),unit_z(2);
+      rot.matrix() << unit_x(0),unit_y(0),unit_z(0),0
+         ,unit_x(1),unit_y(1),unit_z(1),0
+         ,unit_x(2),unit_y(2),unit_z(2),0
+         ,0,0,0,0;
```
Comment (marip8) [inline exercises/Descartes_Planning_and_Execution/solution_ws/src/plan_and_run/src/demo_application.cpp]: This line produces an invalid transform. The last row of a homogenous 4x4 transform should be [0, 0, 0, 1].

File: exercises/Descartes_Planning_and_Execution/template_ws/src/plan_and_run/src/demo_application.cpp [hunk 2]
```diff
@@ -202,9 +202,10 @@ bool DemoApplication::createLemniscateCurve(double foci_distance, double sphere_
       unit_y = (unit_z .cross(unit_x)).normalized();
 
       Eigen::Isometry3d rot;
-      rot.matrix() << unit_x(0),unit_y(0),unit_z(0)
-         ,unit_x(1),unit_y(1),unit_z(1)
-         ,unit_x(2),unit_y(2),unit_z(2);
+      rot.matrix() << unit_x(0),unit_y(0),unit_z(0),0
+         ,unit_x(1),unit_y(1),unit_z(1),0
+         ,unit_x(2),unit_y(2),unit_z(2),0
+         ,0,0,0,0;
```
Comment (marip8) [inline exercises/Descartes_Planning_and_Execution/template_ws/src/plan_and_run/src/demo_application.cpp]: Same comment as above

**Review by marip8 — CHANGES_REQUESTED** [review]

**Comment by gavanderhoorn** [comment]
@marip8: you might consider adding the fixup commits to the PR yourself. That way you don't need to ask the OP for these small changes and the PR doesn't get stuck waiting for them.

**Review by marip8 — APPROVED** [review]

**Comment by marip8** [comment]
@ngwk1 thanks for the contribution

**Pull request merged**